### DISCUSSION TITLE
How do I overlay sheared text onto an image?

### DISCUSSION THREAD

**Discussion by SamusAranX** [ImageMagick/ImageMagick#6231] [Help] [answered]
### ImageMagick version

7.1.1-5

### Operating system, version and so on

macOS Ventura 13.3

### Description

I'm trying to put centered and sheared text with an outline onto a premade template image without *also* applying the shear to the template image. I have a `convert` invocation that does the centering and outlining part pretty well, but I'm at a loss regarding where the shear comes in. 
My example template image is just a 700x700 solid #888888 image and the command I run to get unsheared, outlined text is this:

```
convert ./templates/examplegray.png -background transparent -fill white -stroke black -strokewidth 8.00 -font ./fonts/ARIAL.ttf -gravity center -size 700x400 "caption:LOREM IPSUM" -geometry +0+150 -gravity northwest -composite -background transparent -fill white -font ./fonts/ARIAL.ttf -stroke "#00000001" -strokewidth 8.00 -gravity center -size 700x400 "caption:LOREM IPSUM" -geometry +0+150 -gravity northwest -composite test.png
```

The generated image comes out like this:
![test](https://user-images.githubusercontent.com/676069/230121462-c757c951-0e3a-4743-b807-eca55f448ae5.png)

but I want it to look more like this:
![examplesheared](https://user-images.githubusercontent.com/676069/230121615-b8fcd325-d27b-42ed-84e4-8e6cbdcd75bd.png)

The background won't always be a solid color so merely filling the background with a solid color to counteract the shear won't work. I want the background to not be affected by the shear at all.
Is what I want possible using imagemagick?

**Comment by SamusAranX** [comment]
I should add that I'm aware of `-shear`, but inserting `-shear 0x-17` in this command also skews the background image, no matter where I put it.

**Comment by fmw42** [comment]
Create your text on a transparent background as a separate image. Then shear it and composite it over your background image. 

Alternately, use annotate, which can shear the text in place of caption. The dissadvantage is that you have to do each line separately or put a new line character "\n" in your text to separate the two lines.

See https://imagemagick.org/Usage/text/

**Comment by snibgo** [comment, marked as answer]
You use IM v7, but with `convert` which uses the (very old) v6 syntax. I suggest you use `magick` instead to get the much better v7 syntax (and not `magick convert`, which would also use the old syntax).

You don't want the background to shear, but you are developing/testing with a solid gray background, which won't show any shearing. I suggest you use a background that will show shearing, such as a photo.

I split your command into lines to make it easier to understand.
```
convert \
  ./templates/examplegray.png \
  -background transparent -fill white -stroke black -strokewidth 8.00 -font ./fonts/ARIAL.ttf -gravity center \
  -size 700x400 "caption:LOREM IPSUM" \
  -geometry +0+150 -gravity northwest -composite \
  -background transparent -fill white -font ./fonts/ARIAL.ttf -stroke "#00000001" -strokewidth 8.00 -gravity center z
  -size 700x400 "caption:LOREM IPSUM" \
  -geometry +0+150 -gravity northwest -composite \
  test.png
```

You want to apply a shear to just the text. So you can make an image of just the text, and shear that. Then composite the result over the background image. Like this:
```
magick \
  input.png \
  \( \
    -background transparent -fill white -stroke black -strokewidth 8.00 -font ./fonts/ARIAL.ttf -gravity center \
    -size 700x400 "caption:LOREM IPSUM" \
    -background transparent -fill white -font ./fonts/ARIAL.ttf -stroke "#00000001" -strokewidth 8.00 -gravity center \
    -size 700x400 "caption:LOREM IPSUM" \
    -composite \
    -shear 0x-17 \
  \) \
  -geometry +0+150 -gravity northwest -composite \
  test.png
```
![test](https://user-images.githubusercontent.com/33812368/230143047-51fdd0cc-a38e-4993-869b-a31babf1384b.png)

**Reply by SamusAranX** [reply]
I couldn't get this to work at first, but then it dawned on me that my server, where the prod version of my app lives, was stuck on v6 for silly reasons.

I'm still working on upgrading my `magick` incantation generator but this was a huge help, thanks!
